frontend/utilities/dom_utils/index.ts

```diff
@@ -0,0 +1,26 @@
+import { useLayoutEffect, useState } from "react";
+
+/**
+ * This hook checks if an element is truncated and returns a boolean value.
+ */
+// eslint-disable-next-line import/prefer-default-export
+export const useTruncatedElement = <T extends HTMLElement>(
```

Comment:
Nice! Maybe we should put this in the `frontend/hooks` folder instead?

Reply:
ah right sorry i didnt realise we had that directory. Yah i think it makes sense to put there. will make that change.